nmap: edit page (#12444)
tldr-pages/tldr: pages/common/nmap.md

# nmap

> Network exploration tool and security/port scanner.

> > Some features like SYN scan activates only when `nmap` is run with root privileges.

Reviewer: ```suggestion
> Some features (e.g. SYN scan) activate only when `nmap` is run with root privileges.
```

> More information: <https://nmap.org/book/man.html>.

> - Scan (the top 1000) ports of a remote host with various [v]erbosity levels:

Reviewer: ```suggestion
- Scan the top 1000 ports of a remote host with various [v]erbosity levels:
```

`nmap -v{{1|2|3}} {{ip_or_hostname}}`

- Run a ping sweep over an entire subnet or individual hosts very aggressively:

`nmap -T5 -sn {{192.168.0.0/24|ip_or_hostname1,ip_or_hostname2...}}`

- Enable OS detection, version detection, script scanning, and traceroute:

`sudo nmap -A {{ip_or_hostname1,ip_or_hostname2...}}`

- Scan a specific list of ports (use `-p-` for all ports from 1 to 65535):

`nmap -p {{port1,port2...}} {{ip_or_host1,ip_or_host2...}}`

> - Perform service and version detection of the top 1000 ports using default NSE scripts; writing results (`-oA`) to output files:

Reviewer: ```suggestion
- Perform service and version detection of the top 1000 ports using default NSE scripts, writing results (`-oA`) to output files:
```

> - Perform service and version detection of the top 1000 ports using default NSE scripts; writing results (`-oA`) to output files:

Reviewer: Is there any file other than what's specified here (`top-100-ports`)?
```suggestion
- Perform service and version detection of the top 1000 ports using default NSE scripts; writing results (`-oA`) to the output file:
```

Author: @acuteenvy Yes, `-oA` produces 3 output files of different formats (`top-1000-ports.xml`, `top-1000-ports.nmap` and `top-1000-ports.gnmap`) and hence the word "files".

`nmap -sC -sV -oA {{top-1000-ports}} {{ip_or_host1,ip_or_host2...}}`

- Scan target(s) carefully using `default and safe` NSE scripts:

`nmap --script "default and safe" {{ip_or_host1,ip_or_host2...}}`

> - Scan web server running on standard ports 80 and 443 using all available `http-*` NSE scripts:

Reviewer: ```suggestion
- Scan for web servers running on standard ports 80 and 443 using all available `http-*` NSE scripts:
```

`nmap --script "http-*" {{ip_or_host1,ip_or_host2...}} -p 80,443`

> - Attempt IDS/IPS detection by extremely slow scan (`-T0`), decoy source addresses (`-D`), [f]ragmented packets, random data and other methods:

Reviewer: ```suggestion
- Attempt IDS/IPS detection by using an extremely slow scan (`-T0`), decoy source addresses (`-D`), [f]ragmented packets, random data and other methods:
```
